### PR TITLE
Remove "parent" in tag.rb

### LINKED ISSUE #9403: Remove instances of "parent" in tag.rb
Part of #6367

More context https://github.com/publiclab/plots2/issues/6367#issuecomment-810598964

### PR DESCRIPTION
**Pull request by jcads**
Fixes #9403 

Remove:

- [ ] any instance of the word "parent" in tag.rb
- [x] https://github.com/publiclab/plots2/blob/ee46fe0adf85ba4b64fca68cd844fa05d8eeaa42/app/controllers/tag_controller.rb#L473-L486
- [x] https://github.com/publiclab/plots2/blob/c188b9720ba9cd739321f9efcddbe39810bcce60/config/routes.rb#L209
- [x] https://github.com/publiclab/plots2/blob/ee46fe0adf85ba4b64fca68cd844fa05d8eeaa42/app/controllers/tag_controller.rb#L3

Removing the above should cause the following tests to fail, so remove relevant tests in:
- [x] `plots2/test/unit/node_tag_test.rb`
- [x] `plots2/test/functional/tag_controller_test.rb`

***
Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] ask `@publiclab/reviewers` for help, in a comment below

### CODE REVIEW

File: app/controllers/tag_controller.rb [hunk 1]
```diff
@@ -1,6 +1,5 @@
 class TagController < ApplicationController
   respond_to :html, :xml, :json, :ics
-  before_action :require_user, only: %i(create delete add_parent)
```
Comment (jywarren) [inline app/controllers/tag_controller.rb:3]: Hi, actually here I believe we need to remove just the `add_parent` part, but keep the rest of the line!

File: app/controllers/tag_controller.rb [hunk 2]
```diff
@@ -470,21 +469,6 @@ def contributors_index
     render template: 'tag/contributors-index'
   end
 
-  def add_parent
```
Comment (jywarren) [inline app/controllers/tag_controller.rb:473]: This looks great. We may need to remove the corresponding route: https://github.com/publiclab/plots2/blob/c188b9720ba9cd739321f9efcddbe39810bcce60/config/routes.rb#L209

File: app/models/tag.rb [hunk 3]
```diff
@@ -101,14 +101,9 @@ def self.find_nodes_by_type(tagnames, type = 'note', limit = 10)
...
     # .where('term_data.name IN (?) OR term_data.parent in (?)', tagnames, tagnames) # greedily fetch children
-    tags = Tag.where('term_data.name IN (?)', tagnames)
-    parents = Node.where(status: 1, type: type)
-                  .includes(:tag)
-                  .references(:term_data)
-                  .where('term_data.name IN (?)', tags.collect(&:parent))
     order = 'node_revisions.timestamp DESC'
     order = 'created DESC' if type == 'note'
-    Node.where('node.nid IN (?)', (nodes + parents).collect(&:nid))
+    Node.where('node.nid IN (?)', nodes.collect(&:nid))
```
Comment (jywarren) [inline app/models/tag.rb:106]: Nicely done here! Actually we should be able to remove the comment on line 103 too, also related to parenting:

>     # .where('term_data.name IN (?) OR term_data.parent in (?)', tagnames, tagnames) # greedily fetch children

File: app/models/tag.rb [hunk 4]
```diff
@@ -342,13 +333,13 @@ def self.tagged_nodes_by_author(tagname, user_id)
     if tagname[-1..-1] == '*'
       @wildcard = true
       Node.includes(:node_tag, :tag)
-          .where('term_data.name LIKE(?) OR term_data.parent LIKE (?)', tagname[0..-2] + '%', tagname[0..-2] + '%')
+          .where('term_data.name LIKE(?)', tagname[0..-2] + '%')
```
Comment (jywarren) [inline app/models/tag.rb:336]: Nice, so much simpler!

File: config/routes.rb [hunk 5]
```diff
@@ -206,7 +206,6 @@
   get 'tags/related(/:id)' => 'tag#related'
   get 'tags/:id' => redirect('/tag/:id')
   post 'tag/suggested/:id' => 'tag#suggested'
-  get 'tag/parent' => 'tag#add_parent'
```
Comment (jywarren) [inline config/routes.rb:209]: OH, i see you did it, great, ignore above!

**Review by jywarren — CHANGES_REQUESTED** [review]
Gosh, love seeing all this code go away! 

I think i found a small mistake at `before_action` and suggested a fix. I hope that resolves the tests but we can look closer at what goes wrong in the tests if not! 

Thank you very much and apologies for the slow review. Please feel free to highlight PRs that need review in the chatroom at https://gitter.im/publiclab/publiclab if we miss them, and thanks for your patience!!

File: app/controllers/tag_controller.rb (hunk 1 above)
Comment (jcads) [reply app/controllers/tag_controller.rb:3]: Oh ok, got it!

**Comment by jcads** [comment]
I don't know why there are conflicts though. It seems like it was just the code I removed.

**Comment by RuthNjeri** [comment]
@jcads, the conflict seems related to the tests https://github.com/publiclab/plots2/pull/9404/conflicts
Screenshot 2021-04-15 at 22 19 56

**Review by jywarren — APPROVED** [review]

**Comment by jywarren** [comment]
Awesome work, thank you so much!!!!

**Pull request merged**